### ISSUE TITLE
GET request to a blueprint with an unknown finder returns a 500 server error

### ISSUE THREAD

**Issue by kevinburkeshyp** [balderdashy/sails#2829]
If you expose a database via a blueprint route, and you attempt to filter by a key that is not the name of a column, Sails returns a 500 server error. I reproduced this with Sails 0.11.0 and the Postgres client.

In addition, it seems bad to be setting caching headers on a request that returns a 500 server error.

```
GET /blueprint-model?foo=bar
Host: localhost:1337
```

Response (in development)

```
HTTP/1.1 500 Internal Server Error
Connection: keep-alive
Content-Length: 304
Content-Type: application/json; charset=utf-8
Date: Fri, 10 Apr 2015 17:27:12 GMT
ETag: W/"130-3714686407"
Vary: Accept-Encoding

{
    "error": "E_UNKNOWN",
    "raw": {
        "code": "42703",
        "file": "parse_relation.c",
        "length": 104,
        "line": "2639",
        "name": "error",
        "position": "150",
        "routine": "errorMissingColumn",
        "severity": "ERROR"
    },
    "status": 500,
    "summary": "Encountered an unexpected error"
}
```

Stack trace:

```
error: Sending 500 ("Server Error") response:
 Error (E_UNKNOWN) :: Encountered an unexpected error
error: column blueprintroute.foo does not exist
  at [object Object].Connection.parseE (.../node_modules/sails-postgresql/node_modules/pg.js/lib/connection.js:526:11)
  at [object Object].Connection.parseMessage (.../node_modules/sails-postgresql/node_modules/pg.js/lib/connection.js:356:17)
  at Socket.<anonymous> (.../node_modules/sails-postgresql/node_modules/pg.js/lib/connection.js:105:22)
  at Socket.emit (events.js:95:17)
  at Socket.<anonymous> (_stream_readable.js:764:14)
  at Socket.emit (events.js:92:17)
  at emitReadable_ (_stream_readable.js:426:10)
  at emitReadable (_stream_readable.js:422:5)
  at readableAddChunk (_stream_readable.js:165:9)
  at Socket.Readable.push (_stream_readable.js:127:10)
  at TCP.onread (net.js:528:21)
```

It's a general rule of server development that a bad input should not cause the server to error. I would suggest either ignoring the bad filter or returning a 400 Bad Request instead.

**Comment by crh3675** [comment]
@kevinburkeshyp I agree that the behavior should be more friendly and informative for such errors.  A 500 is pretty serious for a missing column/key error; especially since a user can add arbitrary querystring variables that may consistently result in that sort of error.  You can however, probably catch that error  and handle more gracefully within api/responses.

**Comment by kevinburkeshyp** [comment]
Maybe I am missing something but isn't the whole point of blueprint routes to avoid writing code? Or is there a way to do it ?

FWIW, we're ripping all of the blueprint routes out of our app.

**Comment by CWyrtzen** [comment]
Is there any update on this?

**Comment by sailsbot** [comment]
Thanks for posting, @kevinburkeshyp.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
